tests/Configurator/CopyFromRecipeConfiguratorTest.php

```diff
@@ -190,6 +190,84 @@ public function testUpdate()
         $this->assertSame($newRecipeFiles, $recipeUpdate->getNewFiles());
     }
 
+    public function testUpdateResolveDirectories()
+    {
+        $configurator = $this->createConfigurator();
+
+        $lock = $this->createMock(Lock::class);
+        $lock->expects($this->once())
+            ->method('add')
+            ->with(
+                'test-package',
+                [
+                    'files' => [
+                        'config/packages/framework.yaml',
+                        'test.yaml',
+                    ],
+                ]
+            );
+
+        $originalRecipeFiles = [
+            'symfony8config/packages/framework.yaml' => 'before',
+            'root/test.yaml' => 'before',
+        ];
+        $newRecipeFiles = [
+            'symfony8config/packages/framework.yaml' => 'after',
+            'root/test.yaml' => 'after',
+        ];
+
+        $originalRecipeFileData = [];
+        foreach ($originalRecipeFiles as $file => $contents) {
+            $originalRecipeFileData[$file] = ['contents' => $contents, 'executable' => false];
+        }
+
+        $newRecipeFileData = [];
+        foreach ($newRecipeFiles as $file => $contents) {
+            $newRecipeFileData[$file] = ['contents' => $contents, 'executable' => false];
+        }
+
+        $originalRecipe = $this->createMock(Recipe::class);
+        $originalRecipe->method('getName')
+            ->willReturn('test-package');
+        $originalRecipe->method('getFiles')
+            ->willReturn($originalRecipeFileData);
+
+        $newRecipe = $this->createMock(Recipe::class);
+        $newRecipe->method('getFiles')
+            ->willReturn($newRecipeFileData);
+
+        $recipeUpdate = new RecipeUpdate(
+            $originalRecipe,
+            $newRecipe,
+            $lock,
+            FLEX_TEST_DIR
+        );
+
+        $configurator->update(
+            $recipeUpdate,
+            [
+                'root/' => '',
+                'symfony8config/' => '%CONFIG_DIR%/',
```

Comment:
Having the recipe using `symfony8config` and mapping it to the usual config dir named `config` is not at all the use case that you want to solve (it is exactly the opposite use case). I suggest making the test cover the intended use case instead, as that's the use case for which we want to prevent regressions.

Reply:
I wanted to test, when the path is empty or you mapped to it %SOME-VAR%

I can remove it if you want